go/ct/gen/transaction_context.go

```diff
@@ -19,34 +23,147 @@ import (
 )
 
 type TransactionContextGenerator struct {
+	unsatisfiable bool
+	// This map is used to keep track of variables that are required to have a value
+	// that can be used as index for the blob hash list.
+	blobHashVariables map[Variable]bool
 }
 
 func NewTransactionContextGenerator() *TransactionContextGenerator {
 	return &TransactionContextGenerator{}
 }
 
-func (*TransactionContextGenerator) Generate(rnd *rand.Rand) (*st.TransactionContext, error) {
-	originAddress := common.RandomAddress(rnd)
+func (t *TransactionContextGenerator) Generate(assignment Assignment, rnd *rand.Rand) (*st.TransactionContext, error) {
+	if t.unsatisfiable {
+		return nil, ErrUnsatisfiable
+	}
+
+	maxMustHaveBlobHash := uint64(0)
+	minMustNotHaveBlobHash := uint64(math.MaxUint64)
+
+	// process all the bounded variables to find the maximum present and minimum absent values
+	for variable, hasBlobHash := range t.blobHashVariables {
+		if assignedValue, isBound := assignment[variable]; isBound {
+			if hasBlobHash {
+				if !assignedValue.IsUint64() {
+					return nil, ErrUnsatisfiable
+				}
+				if assignedValue.Uint64() > maxMustHaveBlobHash {
+					maxMustHaveBlobHash = assignedValue.Uint64()
+				}
+			} else {
+				if assignedValue.IsUint64() && assignedValue.Uint64() < minMustNotHaveBlobHash {
+					minMustNotHaveBlobHash = assignedValue.Uint64()
+				}
+			}
+		}
+	}
+
+	if maxMustHaveBlobHash > minMustNotHaveBlobHash ||
+		maxMustHaveBlobHash == math.MaxUint64 ||
+		minMustNotHaveBlobHash == 0 {
```

Comment:
Minimum absent blob hash ==0 is valid if not present blobs hash are required, isn't it? 
I guess the intent is to generate always at least 1 element, but if the condition asks for a non-existing hash at position 0 ?